presto-main/src/main/java/com/facebook/presto/split/PageSinkManager.java

```diff
@@ -64,7 +64,7 @@ public ConnectorPageSink createPageSink(Session session, InsertTableHandle table
     private ConnectorPageSinkProvider providerFor(ConnectorId connectorId)
     {
         ConnectorPageSinkProvider provider = pageSinkProviders.get(connectorId);
-        checkArgument(provider != null, "No page sink provider for connector '%s'", connectorId);
+        checkArgument(provider != null, "No page sink provider for catalog '%s'", connectorId);
```

Comment:
This should probably use `connectorId.getCatalogName()` for the message. @dain should take a look since he recently changed all the connector ID handling.

Reply:
Applied

Reply:
@dain , maybe the `ConnectorId` should be renamed to `CatalogId`?

Reply:
I think it would be even better if we would rename it to `CatalogName`. Also it would be nice if it would be available in SPI so it could be used places like: `com.facebook.presto.spi.connector.ConnectorFactory#create`

Reply:
It has been a long while since I hacked on this, but I believe the conclusion last time was that connectors should get catalog name directly (which they effectively do already).  The reason we used connector id in the first place was so that, we could eventually support MVCC `create catalog` and `drop catalog`, but I don't think that is possible anymore.  The big problem is connectors need to register into things like JMX with consistent well defined names, so that monitoring and alerting systems will work.  Once we accept that restriction, there is no way to have to versions of a catalog active at the same time, so we will need locking to prevent multiple active versions of a catalog.